drivers/serial/uart_ns16550_port_x.h

```diff
@@ -114,7 +114,11 @@ static void irq_config_func_@NUM@(const struct device *dev)
 	/* not PCI(e) */
 
 	IRQ_CONNECT(DT_INST_IRQN(@NUM@),
+#if DT_IRQ_HAS_CELL(@NUM@, priority)
 		    DT_INST_IRQ(@NUM@, priority),
+#else
+		    1,
+#endif
 		    uart_ns16550_isr,
```

Comment:
This is FYI...

For SiFive PLIC, now PLIC supported interrupt priority (by #30238).
So this is not needed for PLIC. But I do not think this commit should be removed if you have some plan to use this mechanism for other simple interrupt-controllers.

Reply:
Thanks for the info for your work on #30238.

The `.dtsi` in this PR has now been updated according to the new PLIC definition. A positive side effect is that the NS16550 change is no longer needed so it has been removed for the PR.